hips/hip-0012.md

```diff
@@ -1,38 +1,141 @@
+- Requiring a timeboxed development effort intends to promote focus on the most valuable changes that the community and maintainers can deliver within a reasonable timeframe.
+It also aims to prevent an "open-ended" development cycle that might drag on or fizzle out.
+Ideally, the majority of changes will be proposed and agreed upon during the planning phase of the development cycle.
+However, amendments to the initial scope are permitted as long as they can fit within the agreed-upon timeframe.
+Explicitly allowing for scope amendments ensures that any changes are well thought out.
+For changes that don't make it into Helm 4's initial development cycle, they can included in Helm 4.x feature releases (subject to minor release compatibility guarantees).
+And for further changes, there will be a Helm 5 release of course!
 
 ## Specification
 
+### The process / timelines
+
+Helm 4 development will consist of three main phases:
+1. Planning
+1. Feature development
+1. Release preparation
+
+The timeline will be approximately:
+
+1. Marking this HIP as "accepted"
+1. Naming of a Release Engineer: Nov. 2024
+1. Kick-off: KubeCon/CloudNativeCon North America 2024 (Nov. 2024, Salt Lake City)
+1. Earliest that engineering can begin: Nov. 2024
+1. Feature freeze for release preparation: By Aug. 2025
+1. Expected release of Helm v4.0.0: By the week before KubeCon/CloudNativeCon North America 2025 (expected November 2025)
+
+Helm v4.1.0 release will be 3 months later (expected Jan. 2026 for the above timeline). And from then on Helm 4 minor releases will follow the same release cadence as Helm v3, shadowing Kubernetes by approximately 1 month.
+
+Helm 3 will reach end of life 6-8 months after Helm 4 release (July 2026 estimated)
+
+### Change suitability
+
+The Helm ecosystem is large, including not just the Helm CLI but also the SDK, Charts, and Chart repositories, plugins.
+Helm also has several [roles or personas](https://github.com/helm/community/blob/main/user-profiles.md) for different user categories, from "Application Operators" to "Helm Developers."
+The Helm project must be careful in how it delivers changes without causing significant adverse impact to Helm's users or the overall Helm ecosystem.
+
+Therefore, while Helm 4 does not need to be semantically backwards compatible with Helm 3, it must not cause significant breakage for users who adopt Helm 4.
+Clear migration options and paths should be documented and presented for Helm 3 users to update to Helm 4.
+
+### Change proposals
+
+To ensure that Helm 4 changes align with this HIP, Helm core maintainers will review proposals.
+These proposals can take the form of lighter-weight Helm 4-specific proposals "H4HIP"s, or standard HIPs suitable for Helm 4 implementation.
+Once a H4HIP is deemed suitable, it will be marked as implementable in the `hips/helm-v4` directory.
```

Comment:
Why the need for the `hips/helm-v4` directory? The current `archives` directory is due to design docs that predate HIPS. Can we just put them all in the `hips` directory?

Reply:
The thought was, that as these are "reduced" HIPs specific to Helm 4. We would want to keep them separate from more general purpose (for lack of a better phrase) HIPs

Reply:
Why keep them separate? Also, the directory makes them seem as though all v4 HIPS should go there. Won't people find the UX weird where full HIPS for v4 are in the `hips` directory?

If I look at the `MUST` call outs in the requirements for a HIP there is only one related to this and that's backwards compat. Since this is a major version jump, little to nothing needs to go there.

What am I missing?

Reply:
decision was made to move "Helm 4 HIPs" to main HIP directory. Changes remaining for "Helm 4 HIPs" are basically to make them as free-form (reduced) as possible compared to normal.